src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderLabel.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Versioning;
+
+namespace System.Security.Cryptography.Cose
+{
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    [RequiresPreviewFeatures(CoseMessage.PreviewFeatureMessage)]
+    public readonly struct CoseHeaderLabel : IEquatable<CoseHeaderLabel>
+    {
+        internal string LabelName => LabelAsInt32?.ToString() ?? LabelAsString ?? 0.ToString();
```

Comment:
Why "0"?  But since it's unreachable, I'd just `!` it away.

```suggestion
        internal string LabelName => LabelAsInt32?.ToString() ?? LabelAsString!;
```

Reply:
It can be reached with `default(CoseHeaderLabel)`, which I want to treat as `new CoseHeaderLabel(0)`.

Reply:
(It's not strictly true for `default(CoseHeaderLabel)`, but it's close enough to true.... or you could just make the property claim it might return null)

Reply:
But if it returns null then the debugger display will show `Label = , Type = Int32`. I would prefer it showing `Label = 0, ...`

Reply:
I believe that the debugger display for the default will claim it's a string, though, so that's not quite consistent.

If you change it to be a `string?` field and an `int` field (non-nullable) and just reverse the flows then a) it'll be smaller on 32-bit builds, and b) `default(CoseHeaderLabel)` and `new CoseHeaderLabel(0)` would be so much the same that you couldn't detect it.

Reply:
```
private string DebuggerDisplay => $"Label = {LabelName}, Type = {(LabelAsInt32.HasValue ? typeof(int) : typeof(string))}";
internal string LabelName => LabelAsInt32?.ToString() ?? LabelAsString ?? 0.ToString();
```

For `default`, LabelAsInt32.HasValue is false, so `DebuggerDisplay` says System.String.

`Label = 0, Type = String`., which would then suggest it encodes as a tstr, which is misleading.

Reply:
Nice idea of reverse the flows.